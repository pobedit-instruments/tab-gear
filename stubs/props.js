import { bindActionCreators } from 'redux';

import actual from './tabs';
import * as Actions from '../views/actions';
import * as ActionTypes from '../views/constants/ActionTypes';
import config from '../config';

export const tabs = {
	actual,
	masked: false,
	active: actual.find(tab => tab.active),
	search: actual
};

export const type = ActionTypes.SHOW_TABS;

export const actions = new function () {
	let actions = {};

	for (let [name, value] of Object.entries(Actions)) {
		actions[name] = bindActionCreators(value, () => {});
	}

	return actions;
};

export { config };
