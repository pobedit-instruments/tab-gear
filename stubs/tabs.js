export default [
	{
		title : 'Google Cloud Computing, Hosting Services & APIs | Google Cloud Platform',
		url   : 'https://cloud.google.com',
		favIconUrl: 'https://cloud.google.com/_static/cc7442d8c7' +
					'/images/cloud/icons/favicons/favicon.ico',
		id : 0,
		index : 0,
		active: true,
		highlighted: false,
		incognito: false
	},

	{
		title : 'Google Plus +++++++++++++++++++++++++++++++++++++++++++',
		url   : 'https://developers.google.com/+/',
		favIconUrl: 'https://developers.google.com/site-assets/logo-google-plus-white.svg',
		id : 1,
		index : 1,
		active: false,
		highlighted: false,
		incognito: false
	},

	{
		title : 'Google Play',
		url   : 'https://play.google.com',
		favIconUrl: 'https://developers.google.com/_static/cc7442d8c7/images/favicon.png',
		id : 2,
		index : 2,
		active: false,
		highlighted: false,
		incognito: false
	},

	{
		title : 'Google Maps',
		url   : 'https://maps.google.com',
		favIconUrl: 'https://www.google.com/images/branding/product/ico/maps_32dp.ico',
		id : 3,
		index : 3,
		active: false,
		highlighted: false,
		incognito: true
	},

	{
		title : 'Google Translate',
		url   : 'https://translate.google.ru/',
		favIconUrl: 'https://translate.google.ru/favicon.ico',
		id : 4,
		index : 4,
		active: false,
		highlighted: false,
		incognito: false
	},

	{
		title : 'Android',
		url   : 'https://developer.android.com',
		favIconUrl: 'https://developer.android.com/favicon.ico',
		id : 5,
		index : 5,
		active: false,
		highlighted: false,
		incognito: false
	},

	{
		title : 'Google Chrome',
		url   : 'https://google.com/chrome',
		favIconUrl: 'https://www.google.com/images/icons/product/chrome-32.png',
		id : 6,
		index : 6,
		highlighted: true,
		active: false
	},

	{
		title : 'Google Developers',
		url   : 'https://developers.google.com/web/',
		favIconUrl: 'https://developers.google.com/web/images/web-fundamentals-icon192x192.png',
		id : 7,
		index : 7,
		active: false,
		highlighted: false,
		incognito: false
	},

	{
		title : 'Google Firebase',
		url   : 'https://firebase.google.com/',
		favIconUrl: 'https://firebase.google.com/_static/cc7442d8c7/images/firebase/favicon.png',
		id : 9,
		index : 9,
		highlighted: false,
		active: false
	}
];
