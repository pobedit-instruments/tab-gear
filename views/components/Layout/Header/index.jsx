import React, { Component, PropTypes } from 'react';
import { bind } from 'decko';
import * as ActionTypes from '../../../constants/ActionTypes';

import Link from '../../Link';

class Header extends Component {
	constructor (properties) {
		super(...arguments);
	}

	/**
	 * Get initial dialog state
	 *
	 * @returns {boolean}
	 */
	ignoreDialog () {
		let state = window.localStorage.getItem('tg-option-dialog');

		try {
			return JSON.parse(state);
		}
		catch (error) {
			return false;
		}
	}

	/**
	 * Close all tabs
	 *
	 * @param {Event} event
	 */
	@bind
	closeAllTabs (event) {
		let { tabs, actions } = this.props;
		let { actual } = tabs;

		if (this.ignoreDialog()) {
			actions.Layout.closeAllTabs(actual);
			actions.Tabs.showTabs(actual);
		}
		else {
			actions.Layout.showDialog(ActionTypes.CLOSE_ALL_TABS, actual);
		}

		event.preventDefault();
	}

	/**
	 * Back to the tabs
	 *
	 * @param {Event} event
	 */
	@bind
	showTabs (event) {
		let { tabs, actions } = this.props;

		actions.Tabs.showTabs(tabs.actual);
		event.preventDefault();
	}

	/**
	 * Show the back link?
	 *
	 * @returns {boolean}
	 */
	hasBack () {
		let types = [
			ActionTypes.SHOW_CREDENTIALS,
			ActionTypes.CLOSE_ALL_TABS
		];

		return types.includes(this.props.type);
	}

	/**
	 * Show the close link?
	 *
	 * @returns {boolean}
	 */
	hasClose () {
		return this.props.type !== ActionTypes.CLOSE_ALL_TABS;
	}

	render () {
		let closeMods = ['close'];

		if (this.props.type === ActionTypes.CLOSE_ALL_TABS) {
			closeMods.push('hide');
		}

		return <div className="tg-panel">
					<Link onClick={ this.showTabs } filter={ this.hasBack() }
					      mods={[ 'back' ]} index="2">
						{ chrome.i18n.getMessage('back') }
					</Link>

					<Link onClick={ this.closeAllTabs } filter={ this.hasClose() }
					      mods={ closeMods } index="3">
						{ chrome.i18n.getMessage('close_all') }
					</Link>
				</div>;
	}
}

Header.propTypes = {
	type   : PropTypes.string.isRequired,
	tabs   : PropTypes.object.isRequired,
	actions: PropTypes.object.isRequired
};

export default Header;
