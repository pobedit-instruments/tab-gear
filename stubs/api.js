import chrome from 'chrome-stub';

import tabs from './tabs';
import locale from '../locale/en/messages.json';

chrome.i18n._locales = locale;

let methods = [
	'query',
	'remove',
	'move',
	'create',
	'discard',
	'highlight'
];

for (let method of methods) {
	chrome.tabs[method].yields(tabs);
}

export default chrome;
