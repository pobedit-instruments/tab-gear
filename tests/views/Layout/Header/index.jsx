import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow, mount, render } from 'enzyme';

import Header from '../../../../views/components/Layout/Header';
import * as props from '../../../../stubs/props';

test('Layout::Header view', () => {
	let component = renderer.create(<Header { ...props } />);
	let view = component.toJSON();

	let actual = expect(view);

	actual.toMatchSnapshot();
});
