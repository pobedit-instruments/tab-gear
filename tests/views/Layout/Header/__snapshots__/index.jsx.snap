exports[`test Layout::Header view 1`] = `
<div
  className="tg-panel">
  <a
    className="tg-link tg-link--close"
    href="#"
    onClick={[Function]}
    tabIndex="3">
    Close all tabs
  </a>
</div>
`;
